Extensions/DialogueTree/dialoguetools.js

```diff
@@ -0,0 +1,301 @@
+/**
+ * @memberof gdjs
+ * @class dialoguetree
+ * @static
+ * @private
+ */
+
+gdjs.dialoguetree = {};
+gdjs.dialoguetree.runner = new bondage.Runner();
+
+/**
+ * Load the Dialogue Tree data of the game.
+ * @param {string} sceneVar The path where to save the screenshot
+ */
+gdjs.dialoguetree.loadFromSceneVar = function(runtimeScene, sceneVar, startDialogueNode) {
+	this.runner = gdjs.dialoguetree.runner; //TODO needs to be initiated globally once, outside of any methods
+	this.yarnData = JSON.parse(sceneVar.getAsString());
+	this.runner.load(this.yarnData);
+
+	if (startDialogueNode && startDialogueNode.length > 0) {
+		gdjs.dialoguetree.startFrom(startDialogueNode);
+	}
+};
+
+gdjs.dialoguetree.startFrom = function(startDialogueNode) {
+	this.optionsCount = 0;
+	this.options = [];
+	this.dialogueIsRunning = true;
+	this.dialogueBranchTitle = '';
+	this.dialogueBranchBody = '';
+	this.dialogueBranchTags = [];
+	this.dialogue = this.runner.run(startDialogueNode);
+	this.NextDialogueData = null;
```

Comment:
nitpicking: `this.nextDialogueData` (no capital letter).

Also, would be great to move all the private variables to a name starting with an underscore. Don't necessarily do it now, but will be useful for maintenance later.

Reply:
I decided against storing the dialogue data in two objects so this no longer applies :)

Reply:
ah sry,responded  to wrong comment xD the first d or the second d?